### PR TITLE
Add CI that checks everything nightly

### LINKED ISSUE #46: read_console_input read() error handling seems to no longer be correct.
Hey!
When running read_console_input.odin example, the following error gets thrown:
```
Error: Cannot compare expression, operator '<' not defined between the types 'Error' and 'untyped nil'
	if err < 0 {
```
I have an old small odin project that uses the same handling and noticed it didn't work anymore, would you know how to best handle read() error returns now?

Thanks :)

edit: handling it the following way for now, curious to hear if there's a different/more idiomatic way of doing it
```
if err != nil {
```

### PR DESCRIPTION
**Pull request by laytan**
Fixes #46

### CODE REVIEW

File: by_example/dir_info/main.odin [hunk 1]
```diff
@@ -6,49 +6,50 @@ import "core:path/filepath"
...
-    }
-}
\ No newline at end of file
+	cwd := os.get_current_directory()
+	fmt.println("Current working directory:", cwd)
+
+	f, err := os.open(cwd)
+	defer os.close(f)
+
+	if err != os.ERROR_NONE {
```
Comment (gingerBill) [inline by_example/dir_info/main.odin:36]: When the next release happens, we can probably make this `err != nil`. But this is GREAT for now.

File: raylib/game_of_life/game_of_life.odin [hunk 2]
```diff
@@ -70,190 +70,185 @@ User_Input :: struct {
...
-    }
+	for x: i32 = 0; x < world.width; x += 1 {
+		for y: i32 = 0; y < world.height; y += 1 {
+			neighbors := count_neighbors(world, x, y)
+			index := y * world.width + x
+
+			switch neighbors {
+			case 2: {next_world.alive[index] = world.alive[index]}
+			case 3: {next_world.alive[index] = 1}
+			case:   {next_world.alive[index] = 0}
```
Comment (gingerBill) [inline raylib/game_of_life/game_of_life.odin:81]: Why are the `{}` here in the first place?

File: raylib/game_of_life/game_of_life.odin [hunk 3]
```diff
@@ -70,190 +70,185 @@ User_Input :: struct {
...
+	bottom_left  := w.alive[down * w.width + left]
+	bottom       := w.alive[down * w.width + x]
+	bottom_right := w.alive[down * w.width + right]
 
-    top_row    := top_left    + top     + top_right
-    mid_row    := mid_left              + mid_right
-    bottom_row := bottom_left + bottom  + bottom_right
+	top_row    := top_left + top + top_right
+	mid_row    := mid_left + mid_right
+	bottom_row := bottom_left + bottom + bottom_right
```
[the remark this answers is not in the document]
Comment (laytan) [reply raylib/game_of_life/game_of_life.odin:111]: Oh right, let me fix that

File: raylib/microui/microui_raylib_demo.odin [hunk 4]
```diff
@@ -179,11 +179,13 @@ render :: proc "contextless" (ctx: ^mu.Context) {
 		switch cmd in variant {
 		case ^mu.Command_Text:
 			dst := rl.Rectangle{f32(cmd.pos.x), f32(cmd.pos.y), 0, 0}
-			for ch in cmd.str do if ch&0xc0 != 0x80 {
-				r := min(int(ch), 127)
-				src := mu.default_atlas[mu.DEFAULT_ATLAS_FONT + r]
-				render_texture(state.screen_texture, &dst, src, to_rl_color(cmd.color))
-				dst.x += dst.width
+			for ch in cmd.str {
```
Comment (gingerBill) [inline raylib/microui/microui_raylib_demo.odin:182]: This is not related to this specific code but be should probably add support for the new `core:text/edit` integration to `vendor:microui`.

File: wgpu/microui/os_glfw.odin [hunk 5]
```diff
@@ -13,146 +13,146 @@ import     "vendor:wgpu/glfwglue"
...
-    case glfw.PRESS, glfw.REPEAT: mu.input_key_down(&state.mu_ctx, mu_key)
-    case glfw.RELEASE:            mu.input_key_up  (&state.mu_ctx, mu_key)
-    case:                         return
-    }
+	context = state.ctx
+
+	mu_key: mu.Key; switch key {
+	case glfw.KEY_LEFT_SHIFT, glfw.KEY_RIGHT_SHIFT:     mu_key = .ALT
+	case glfw.KEY_LEFT_CONTROL, glfw.KEY_RIGHT_CONTROL,
+		glfw.KEY_LEFT_SUPER, glfw.KEY_RIGHT_SUPER:      mu_key = .CTRL
```
Comment (gingerBill) [inline wgpu/microui/os_glfw.odin:94]: Alignment should be spaces here. 
```odin
	case glfw.KEY_LEFT_CONTROL, glfw.KEY_RIGHT_CONTROL,
	     glfw.KEY_LEFT_SUPER, glfw.KEY_RIGHT_SUPER:      mu_key = .CTRL
```

File: raylib/microui/microui_raylib_demo.odin (hunk 4 above)
Comment (laytan) [reply raylib/microui/microui_raylib_demo.odin:182]: I think it already is supported here, I can copy, paste etc.

**Pull request merged**